internal/pkg/cli/env_delete.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// DeleteEnvOpts holds the fields needed to delete an environment.
+type DeleteEnvOpts struct {
+	Env              string
+	SkipConfirmation bool
+}
+
+// Ask is a no-op for this command.
+func (opts *DeleteEnvOpts) Ask() error {
+	return nil
+}
+
+// Validate returns an error if the environment name does not exist in the project, or there are existing applications
+// in the environment, or the environment is not available using the EnvironmentManagerRole.
+func (opts *DeleteEnvOpts) Validate() error {
+	return nil
+}
+
+// Execute deletes the environment from the project by first deleting the stack and then removing the entry from the store.
+func (opts *DeleteEnvOpts) Execute() error {
+	return nil
+}
+
+// RecommendedActions is a no-op for this command.
+func (opts *DeleteEnvOpts) RecommendedActions() []string {
+	return nil
+}
+
+// BuildEnvDeleteCmd builds the command to delete environment(s).
+func BuildEnvDeleteCmd() *cobra.Command {
+	opts := &DeleteEnvOpts{}
+	cmd := &cobra.Command{
+		Use:   "delete [name]",
+		Short: "Deletes an environment from your project.",
+		Example: `
+  Delete the "test" environment.
+  /code $ archer env delete test
+
+  Delete the "test" environment without prompting.
+  /code $ archer env delete test prod --yes`,
```

Comment:
Should we have a "force" flag instead of yes? Here's why I'm thinking about that.

If we delete the stack, but can't delete the parameter, for whatever reason, the parameter will be orphaned. 

If they pass the force flag, we can still delete the parameter, even if we fail to delete the stack / or assume the env role.

Reply:
Interesting, I feel like they accomplish two different things. `--yes` is really a way of acknowledging that the user knows what they are doing. 

I wonder if what you described should be just the default behavior, that's how `rm` works right? If a subdirectory can't be deleted it doesn't halt and keeps moving forward. What do you think?

Reply:
Yea, that makes sense :D